gax-java/gax-httpjson/src/test/java/com/google/api/gax/httpjson/RetryingTest.java

```diff
@@ -319,6 +347,13 @@ void retryKeepFailing() {
 
     UncheckedExecutionException exception =
         assertThrows(UncheckedExecutionException.class, () -> Futures.getUnchecked(future));
+    // the number of attempts varies. Here we just make sure that all of them except the last are
+    // considered as failed
```

Comment:
qq, isn't the last one also considered a failed attempt? Why is the tracerAttemptsFailed != tracerAttempts?

Also, can we add the operationfailed == true check here as well?

Reply:
> Why is the tracerAttemptsFailed != tracerAttempts?

I'm not sure and I agree they should be the same value since this test just fetches an exception until it can't retry anymore (retries exhausted).

Reply:
from discussion with @blakeli0, @lqiu96 and @zhumin8, let's double check `TestApiTracer`, confirm with showcase.

Reply:
showcase has another tracer we can use to cross check `TestApiTracer`